### PR TITLE
Create recipes for PyJulia and PySR

### PR DESCRIPTION
**Pull request by MilesCranmer**
Here's the fixed version of #17605.

cc @ngam @mkitti @ocefpaf @isuruf 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MilesCranmer** [comment]
@conda-forge-admin, please ping team @conda-forge/staged-recipes

**Comment by ngam** [comment]
We will want to make sure this is resolved correctly. This is what it means: https://github.com/conda-forge/pyjulia-feedstock and https://github.com/conda-forge/pysr-feedstock



> Hi! This is the friendly automated conda-forge-linting service.
> 
> I wanted to let you know that I linted all conda-recipes in your PR (`recipes/pyjulia`, `recipes/pysr`) and found some lint.
> 
> Here's what I've got...
> 
> For **recipes/pyjulia**:
> 
> * Feedstock with the same name exists in conda-forge.
> 
> For **recipes/pysr**:
> 
> * Feedstock with the same name exists in conda-forge.

File: recipes/pyjulia/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python
+    - pip
```
Comment (isuruf) [inline recipes/pyjulia/meta.yaml:19]:
```suggestion
```

File: recipes/pysr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python
+    - pip
```
Comment (isuruf) [inline recipes/pysr/meta.yaml:19]:
```suggestion
```

File: recipes/pysr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  host:
+    - python >=3.6,<4.0
+    - pip
+    - pyjulia >=0.5.7
+    - numpy >=1.14.0
+    - pandas
+    - sympy
+    - scikit-learn
```
Comment (isuruf) [inline recipes/pysr/meta.yaml:27]:
```suggestion
```

File: recipes/pysr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  run:
+    - python >=3.6,<4.0
+    - {{ pin_compatible('pyjulia') }}
+    - {{ pin_compatible('numpy') }}
+    - {{ pin_compatible('pandas') }}
+    - {{ pin_compatible('sympy') }}
+    - {{ pin_compatible('scikit-learn') }}
```
Comment (isuruf) [inline recipes/pysr/meta.yaml:34]: Remove use of `pin_compatible` here

File: recipes/pyjulia/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{% set version = "0.5.7" %}
+
+package:
+  name: pyjulia
+  version: {{ version }}
+
+source:
+  - url: https://files.pythonhosted.org/packages/33/90/ba856cddadd923c94b7ce07dd3623d37cfea22b815076f0c70a47d867a3e/julia-{{ version }}.tar.gz
```
Comment (isuruf) [inline recipes/pyjulia/meta.yaml:8]: Use the URL template from example recipe

**Comment by MilesCranmer** [comment]
Thanks @isuruf! All implemented.

**Comment by MilesCranmer** [comment]
Just pinging this. Let me know what is left, thanks. @conda-forge/staged-recipes

**Comment by MilesCranmer** [comment]
@conda-forge-admin, please ping team @conda-forge/staged-recipes

**Comment by mkitti** [comment]
Would it make sense to display next step instructions somewhere late in the install process?

**Comment by MilesCranmer** [comment]
Sounds good to me. Where should I print this? 

It will also display the next step instructions in an error message if the user tries to do anything before running `.install()`.

**Comment by mkitti** [comment]
I'm a newbie here in this regard. I would guess a post-link script, but I know nothing. @ngam ?

**Comment by ngam** [comment]
Yes, you could do post-link, e.g. https://github.com/conda-forge/cudatoolkit-feedstock/blob/master/recipe/post-link.sh. But I think you can drop this until the user tries to do something with the package. 

Think about it this way: the user installs this package to use it, right? and so when they try to use it, they will get a warning about setting up things correctly (that also should be part of the explanation upstream)

**Comment by ngam** [comment]
> It will also display the next step instructions in an error message if the user tries to do anything before running `.install()`.

I think this is enough. We can also think more about this once in a feedstock, as part of a general cohesive push among pkgs using julia

**Comment by MilesCranmer** [comment]
That makes sense to me. I'm good with leaving it as is with the error message.

**Pull request merged**